### PR TITLE
chore: validates t7 driver

### PR DESCRIPTION
**Pull request by maxleblang**
## Summary

Validates the T7 driver against a physical LabJack T7 instrument. Adding on to the test suite from LabJack T4 to cover T7 specific functionalities.

## Type of change

- [ ] Bug fix (`fix`)
- [ ] New feature (`feat`)
- [ ] Breaking change (`feat!` / `fix!`)
- [ ] Refactor (`refactor`)
- [ ] Documentation (`docs`)
- [x] Chore / tooling (`chore`)

## Verification
T7 test passing

annotated T7 data

## Tests

- [x] Unit tests added or updated
- [ ] Existing tests cover this change
- [ ] No tests — explain why:

## Checklist

- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) (e.g. `feat(driver): add support for Keysight E36300`)
- [ ] I have read [CONTRIBUTING.md](CONTRIBUTING.md)
- [ ] Documentation updated if user-facing behavior changed
- [ ] Code follows the style/conventions of the surrounding code

## Notes for reviewers

_Optional — call out specific files, edge cases, or decisions you'd like eyes on._

### CODE REVIEW

**Comment by nhschwab** [comment]
The screenshot of the Nominal workbook looks like it only includes the first tests up until the analog loopback (second two rows are empty). Maybe zoom out so we can get the full picture? Can you also include a screenshot of the console output of the pytest results?

**Comment by nhschwab** [comment]
Can we add tests specific to the T7? Different AI ranges, differential terminal config, multiplexing (add another connection DAC1 -> AIN1)

**Comment by nhschwab** [comment]
Found a bug in the labjack T7 driver while reviewing: https://github.com/nominal-io/instro/blob/main/packages/instro-daq-labjack/instro/daq/drivers/labjack/t_series_models.py#L137-L156

`aValues` should have four zeros to match the length of `aNames`. We were getting lucky with the out-of-bounds process memory returning a near zero value so this went undetected. Can you fix this as part of this PR? Out of scope of original issue but is small enough and benign

File: packages/instro-daq-labjack/instro/daq/drivers/labjack/t_series_models.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def hw_timing_configs(
             "STREAM_RESOLUTION_INDEX",
             "STREAM_SETTLING_US",
         ]
-        aValues = [0, 0, 0]
+        aValues = [0, 0, 0, 0]
```
Comment (maxleblang) [inline packages/instro-daq-labjack/instro/daq/drivers/labjack/t_series_models.py:154]: T7 driver bug fix

File: tests/daq/labjack/test_labjack_t7_hardware.py [hunk 2]
```diff
@@ -0,0 +1,781 @@
...
+            "Digital line loopback",
+            "Drive FIO0 through a 0/1 sequence and verify FIO1 reads back the same state "
+            "via single-line loopback wiring.",
+            step,
+        )
+
+    # =====================================================================
+    # 8. Multichannel HW-timed analog read with background daemon
+    # =====================================================================
+    def test_08_hw_timed_analog_read_background(self):
```
Comment (maxleblang) [inline tests/daq/labjack/test_labjack_t7_hardware.py:594]: We should copy this multi-channel version over to T4

File: tests/daq/labjack/test_labjack_t7_hardware.py [hunk 3]
```diff
@@ -0,0 +1,781 @@
...
+            f"Start multichannel HW-timed acquisition at {SAMPLE_RATE_HZ} Hz with background daemon. "
+            f"Hold DAC0 at {HW_TIMED_DC_V0} V and DAC1 at {HW_TIMED_DC_V1} V; verify AIN0 and AIN1 buffers "
+            "each track their own source via get_channel().",
+            step,
+        )
+
+    # =====================================================================
+    # 9. Multichannel HW-timed analog read without background daemon
+    # =====================================================================
+    def test_09_hw_timed_analog_read_no_background(self):
```
Comment (maxleblang) [inline tests/daq/labjack/test_labjack_t7_hardware.py:649]: Same here. Move over to T4

File: tests/daq/labjack/test_labjack_t7_hardware.py [hunk 4]
```diff
@@ -0,0 +1,781 @@
...
+    (-0.01, 0.01, 0.005, 0.1),
+]
+
+# True when DAC0->AIN0 and FIO0->FIO1 are physically looped back. Gates the
+# strict value checks; structural checks always run.
+LOOPBACK_WIRED = True
+
+# DAC0 spans 0-5 V, so every analog test point stays inside that range.
+ANALOG_TEST_VOLTAGES = [0.0, 0.5, 1.25, 2.5, 3.3, 4.5]
+ANALOG_TOLERANCE_V = 0.05  # DAC ~10 mV + AIN noise/offset; 50 mV is comfortable.
```
Comment (nhschwab) [inline tests/daq/labjack/test_labjack_t7_hardware.py:119]: Since we are testing the T7 ranges, the tolerance voltage of 0.05V is a poor choice for testing the +/- 0.1V and +/- 0.01 V ranges. We should update the tolerance to be dependent on voltage range being tested

File: tests/daq/labjack/test_labjack_t7_hardware.py [hunk 5]
```diff
@@ -0,0 +1,781 @@
...
+
+  Digital loopback (wire FIO0 -> FIO1):
+    FIO0 (driven as output)  --->  FIO1 (read as input)
+
+  Channel configuration summary:
+    AI ch "AIN0"  — alias "ain0", RSE (loopback from DAC0)
+    AI ch "AIN1"  — alias "ain1", RSE (loopback from DAC0)
+    AO ch "DAC0"  — alias "dac0", 0-5 V
+    DO line "FIO0" — alias "fio0", Logic.HIGH
+    DI line "FIO1" — alias "fio1", Logic.HIGH
```
Comment (nhschwab) [inline tests/daq/labjack/test_labjack_t7_hardware.py:37]: Analog wiring docs seem off/inconsistent, can you double check? Missing DAC1 and AIN1 is a loopback from DAC1, not DAC0?

File: uv.lock
Comment (nhschwab) [inline uv.lock:1]: We don't need to bump package versions in this PR

File: tests/daq/labjack/test_labjack_t7_hardware.py [hunk 6]
```diff
@@ -0,0 +1,901 @@
...
+    DAC0 (AO, 0-5 V)  --->  AIN0  (AI)
+    DAC1 (AO, 0-5 V)  --->  AIN1  (AI)
+
+  Digital loopback (wire FIO0 -> FIO1):
+    FIO0 (driven as output)  --->  FIO1 (read as input)
+
+  Channel configuration summary:
+    AI ch "AIN0"  — alias "ain0", RSE (loopback from DAC0)
+    AI ch "AIN1"  — alias "ain1", RSE (loopback from DAC1)
+    AO ch "DAC0"  — alias "dac0", 0-5 V
```
Comment (nhschwab) [inline tests/daq/labjack/test_labjack_t7_hardware.py:34]:
```suggestion
    AO ch "DAC0"  — alias "dac0", 0-5 V
    AO ch "DAC1"  — alias "dac1", 0-5 V
```

File: uv.lock
Comment (LizzyC-115) [reply uv.lock:1]: Reseted the uv.lock file to match main

**Review by nhschwab — APPROVED** [review]
